src/routes/new-reservation/+page.svelte

```diff
@@ -0,0 +1,448 @@
+<script>
+    // Framework imports
+    import { getToastStore } from "@skeletonlabs/skeleton";
+    import { goto } from "$app/navigation";
+
+    // Import backend urls
+    import { urls, tryFetchingRestricted } from "$lib/util.js";
+    import { onMount } from "svelte";
+
+    // Definitions
+
+    //Constants
+    const toastStore = getToastStore();
+
+    // Toast Settings
+    const toastSuccsess = {
+        message: "Reservierung wurde erfolgreich erstellt",
+        hideDismiss: true, // Hide the dismiss button on toast
+        timeout: 3000, // Auto dismiss toast after 3 seconds
+        background: "variant-filled-secondary",
+    };
+
+    const toastError = {
+        message: "Reservierung konnte nicht angelegt werden",
+        hideDismiss: true, // Hide the dismiss button on toast
+        timeout: 3000, // Auto dismiss toast after 3 seconds
+        background: "variant-filled-error",
+    };
+
+    const indicatorStatus = {
+        none: "",
+        warning: "!border-warning-400",
+        error: "!border-error-600",
+    };
+
+    const dateOptions = {
+        year: "numeric",
+        month: "2-digit",
+        day: "2-digit",
+        hour: "numeric",
+        minute: "numeric",
+        hour12: false,
+    };
+
+    // Variables
+    let selectedStationIndex = 0;
+
+    // Global temporary save variables
+    let selectedStation;
+    let selectedBeginn;
+    let selectedEnd;
+    let selectedCar;
+
+    // Indicators
+    let beginIndicator = indicatorStatus.none;
+    let endIndicator = indicatorStatus.none;
+    let periodIndicator = false;
+    let emptyIndicator = false;
+
+    // Binding references
+    let beginReference;
+    let endReference;
+
+    // Formpage index for changing form view according to user reservation progress
+    let formPage = 0;
+
+    // Initialize promises as empty on mount
+    let cars = new Promise((resolve, reject) => {});
+    let stations = new Promise((resolve, reject) => {});
+
+    function resetIndicators() {
+        periodIndicator = false;
+        emptyIndicator = false;
+        beginIndicator = indicatorStatus.none;
+        endIndicator = indicatorStatus.none;
```

Comment:
Gleiche Reihenfolge wie oben
```suggestion
        beginIndicator = indicatorStatus.none;
        endIndicator = indicatorStatus.none;
        periodIndicator = false;
        emptyIndicator = false;
```